addons/account/models/account_move.py

```diff
@@ -1189,8 +1190,14 @@ def _compute_payment_state(self):
                         new_pmt_state = 'partial'
                     elif invoice.matched_payment_ids.filtered(lambda p: not p.move_id and p.state == 'paid'):
                         new_pmt_state = invoice._get_invoice_in_payment_state()
+
+            if new_pmt_state == 'paid' and invoice.payment_state != 'paid':
```

Comment:
What about `in_payment`?

Reply:
The hook would be called by the reconciliation when the invoice is marked as paid